### PR TITLE
Remove deprecated FunctionInfo and FunctionIdent

### PR DESCRIPTION
**Pull request by seut**
## Summary of the changes / Why this improves CrateDB

Since CrateDB 4.2.0, `FunctionInfo` and `FunctionIdent` are superseded by the `Signature` class. We kept both classes for compatibility with older nodes and still streamed them.
This removes theses classes completely while still stream a `FunctionInfo` format for compatibility with these old nodes.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by seut** [comment]
retest this please

unrelated `testRerouteRecovery` test failure

File: server/src/main/java/io/crate/expression/eval/EvaluatingNormalizer.java [hunk 1]
```diff
@@ -174,17 +174,19 @@ public void writeTo(StreamOutput out) throws IOException {
...
                             if (out.getVersion().onOrAfter(Version.V_4_2_0)) {
-                                out.writeBoolean(signature != null);
-                                if (signature != null) {
-                                    signature.writeTo(out);
-                                    DataTypes.toStream(returnType, out);
+                                if (out.getVersion().before(Version.V_5_0_0)) {
+                                    out.writeBoolean(true);
                                 }
+                                signature.writeTo(out);
+                                DataTypes.toStream(returnType, out);
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/expression/eval/EvaluatingNormalizer.java:189]: Aren't some of these after conditions dead code due to the 

```
if (out.getVersion().onOrAfter(Version.V_4_2_4)) {
                            super.writeTo(out);
```

at the top?

**Review by mfussenegger — APPROVED** [review]
Nice.

Left a minor comment, otherwise lgtm

File: server/src/main/java/io/crate/expression/eval/EvaluatingNormalizer.java (hunk 1 above)
Comment (seut) [reply server/src/main/java/io/crate/expression/eval/EvaluatingNormalizer.java:189]: Actually, my added `before 5.0.0` checks can be removed due to that, good catch, thx.
But the other `onOrAfter` must stay as they match  "version <= 4.2.4 && version >= 4.1.0/4.2.0" conditions.

**Pull request merged**